File: Server/src/main/java/com/api/server/foodb/CiteThisExternalLinkReferencing.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "cite_this_external_link_referencings")
public class CiteThisExternalLinkReferencing {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "external_link_id", nullable = false)
    private CiteThisExternalLink externalLink;

    @NotNull
    @Column(name = "referencer_id", nullable = false)
    private Integer referencerId;

    @Size(max = 255)
    @NotNull
    @Column(name = "referencer_type", nullable = false)
    private String referencerType;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

}

File: DotNetServer/DotNetServer/Core/Entities/Foodb/CiteThisExternalLinkReferencing.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Core.Entities.Foodb;

[Table("cite_this_external_link_referencings")]
[Index("ReferencerId", "ReferencerType", Name = "external_link_referencings_all_ids")]
[Index("ExternalLinkId", Name = "index_cite_this_external_link_referencings_on_external_link_id")]
[Index("ReferencerId", Name = "index_cite_this_external_link_referencings_on_referencer_id")]
public partial class CiteThisExternalLinkReferencing
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("external_link_id")] public int ExternalLinkId { get; set; }

    [Column("referencer_id")] public int ReferencerId { get; set; }

    [Column("referencer_type")] public string ReferencerType { get; set; } = null!;

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }

    [ForeignKey("ExternalLinkId")]
    [InverseProperty("CiteThisExternalLinkReferencings")]
    public virtual CiteThisExternalLink ExternalLink { get; set; } = null!;
}